### PR TITLE
fix(a380x/fws): Improve AP&A/THR disco logic; also process quick successions of AP instinctive disc. p/b presses

### LINKED ISSUE #9094: A380X/FWS: ATHR OFF Disconnect logic
### Aircraft Version

Development

### Build info

```json
master
```

### Describe the bug

Setting the thrust levers to toga or MCT while the autothrust was previously active, activates the AUTO FLT ATHR OFF ecam message.
In addition, setting the thrust levers to idle does the same.

### Expected behavior

Setting the levers to toga or MCT should not trigger the condition, in addition, setting the thrust levers to idle should be considered a voluntary disconnect and trigger the ATHR OFF memo as long as RA > 50FT.

### Steps to reproduce

1) Set the thrust levers to the climb detent and engnage ATHR, then, set the thrust levers to toga or mct.
2) When ATHR is engaged, set the thrust levers to idle

### References (optional)

FCOM

### Additional info (optional)

Bonus: Sometimes when disconnecting then activating then disconnecting the ATHR again voluntary in a very short time the memo does not happen nor the chime.

### Discord Username (optional)

bruno_pt99

### LINKED ISSUE #9110: [A380X] Quick double press not captured in FWS for AP disconnect
floridude knows, floridude will fix

### PR DESCRIPTION
**Pull request by flogross89**
Fixes #9094 
Fixes #9110 

## Summary of Changes

- Buffer discrete inputs now, process them in the next FWS cycle. Count AP disco presses if multiple between one FWS cycle
- A/THR logic improvement: Don't show ECAM fault when pulling to idle or setting MAN xxx thrust
- AP logic improvement concerning double presses, don't require delay nodes anymore

## Additional context

Discord username (if different from GitHub):

## Testing instructions

TODO, coming tomorrow

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts [hunk 1]
```diff
@@ -460,15 +471,14 @@ export class FwsCore {
...
-
-  public readonly autoPilotDisengagedDelay = new NXLogicTriggeredMonostableNode(0.1, true);
+  /** If multiple AP discs are triggered between FWS cycles, memorize the amount */
+  public autoPilotInstinctiveDiscCountSinceLastFwsCycle = 0;
 
-  public readonly autoPilotDisengagedDelayedPulse = new NXLogicPulseNode(false);
+  public readonly autoPilotDisengagedInstantPulse = new NXLogicPulseNode(false);
 
-  public readonly autoPilotInstinctiveDiscPressedInLast1p8Sec = new NXLogicTriggeredMonostableNode(1.9, false);
+  public readonly autoPilotInstinctiveDiscPressedInLast1p8Sec = new NXLogicTriggeredMonostableNode(1.9, true);
```
Comment (BlueberryKing) [inline fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts:479]: The variable name doesn't match the 1.9s you give it

Comment (flogross89) [reply fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts]: Thanks for the review. The references say 1.8s, but due to our audio files I had to raise it to 1.9s to that the triple click can finish before the loud CRC. I added a comment, we could maybe touch that again when we have proper triple click logic

**Comment by flogross89** [comment]
**Quality Assurance Report, as proxy to danbw. From discord:** 

Just tested it and it works as it should on A380. 

Pressed twice on the AP instinctive disconnect, cavalry charge stops abrupty with second press along with the ECAM Memo. Also no A/THR off memo when levers set to idle on landing.

Also tried A/THR off with two instinctive disconnects, works same as AP off. 

Slightly different to A320, where the cavalry charge completes one aural alert cycle and memo stays on ECAM a little longer. so I just re-confirmed it via YouTube video of IRL cockpit landing.

**Testing Results: Passed**

File: fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts [hunk 2]
```diff
@@ -2438,24 +2467,23 @@ export class FwsCore {
     this.autoPilotInstinctiveDiscPressedPulse.write(false, deltaTime);
 
     // A/THR OFF
-    const aThrEngaged = this.autoThrustStatus.get() === 2;
+    const aThrEngaged = this.autoThrustStatus.get() === 2 || this.autoThrustMode.get() !== 0;
```
Comment (BlueberryKing) [inline fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts:2470]: What is the purpose of `this.autoThrustMode.get() !== 0`?

**Review by BlueberryKing — APPROVED** [review]

File: fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts (hunk 2 above)
Comment (flogross89) [reply fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsCore.ts:2470]: For some of the A/THR modes such as MAN FLEX, MAN MCT etc. A/THR is armed but not engaged, but we still don't want to trigger the memo or fault. This ensures we only trigger the memo and/or fault if A/THR mode is NONE. Might be partly redundant to Status being 2, but didn't want to change the trigger conditions too much right now

**Pull request merged**